js/blocks/components/fields.js

```diff
@@ -59,33 +60,31 @@ const Fields = ( { fields, parentBlockProps, parentBlock, rowIndex } ) => {
 		 * the block's setAttributes property, so that the block can save the value.
 		 * This function is passed to the control so that the control can save the value,
 		 * depending on whether the control is in a repeater row or not.
+		 * This gets the latest parentAttributes by using select( 'core/block-editor' ),
+		 * as the parentBlockProps.attributes in the outer scope may be outdated.
```

Comment:
It looks like the issue with the disappearing Classic Text fields was that it was using outdated `parentBlockProps.attributes`. So this gets them with the store.